Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectILLReductionDIFF.py

```diff
@@ -0,0 +1,147 @@
+                             doc="Number of pixels to mask at the end of each tube")
+
+        self.declareProperty("Observable", "sample.temperature",
+                             doc="If multiple files, the parameter from SampleLog to use as an index when conjoined.")
+
+    def _normalize_by_monitor(self, ws):
+        """
+            Normalizes the workspace by monitor values (ID is 0 for IN16B)
+            @param ws : the input workspace
+        """
+        monitor_ws = ws + '_mon'
+        ExtractMonitors(InputWorkspace=ws, DetectorWorkspace=ws, MonitorWorkspace=monitor_ws)
+
+        # in case of 0 counts monitors, replace 0s by 1s so the division becomes neutral
+        # (since it's generally division of 0 detector's counts by 0 monitor's counts,
+        # they weren't very useful to begin with)
+
+        ReplaceSpecialValues(InputWorkspace=monitor_ws, OutputWorkspace=monitor_ws, SmallNumberThreshold=0.00000001,
+                             SmallNumberValue=1)
+
+        Divide(LHSWorkspace=ws, RHSWorkspace=monitor_ws, OutputWorkspace=ws, WarnOnZeroDivide=True)
+
+        cache = list(range(1, self.mask_start_pixels)) + list(range(257 - self.mask_end_pixels, 257))
+        to_mask = [i + 256 * j for i in cache for j in range(8)]
+        MaskDetectors(Workspace=ws, DetectorList=to_mask)
+        DeleteWorkspace(monitor_ws)
+
+    def _treat_doppler(self, ws):
+        """
+            Reduce Doppler diffraction data presents in workspace.
+            @param ws: the input workspace
+        """
+        run = None
+        if len(self.runs) > 1:
+            number_of_channels = mtd[mtd[ws].getNames()[0]].blocksize()
+            run = mtd[mtd[ws].getNames()[0]].getRun()
+        else:
+            number_of_channels = mtd[ws].blocksize()
+            run = mtd[ws].getRun()
+
+        if run.hasProperty('Doppler.incident_energy'):
+            energy = run.getLogData('Doppler.incident_energy').value / 1000
+        else:
+            raise RuntimeError("Unable to find incident energy for Doppler mode")
+
+        Rebin(InputWorkspace=ws, OutputWorkspace=self.output, Params=[0, number_of_channels, number_of_channels])
+        self._normalize_by_monitor(self.output)
+
+        ConvertSpectrumAxis(InputWorkspace=self.output,
+                            OutputWorkspace=self.output,
+                            Target='ElasticQ',
+                            EMode="Direct",
+                            EFixed=energy)
+
+        ConvertToPointData(InputWorkspace=self.output, OutputWorkspace=self.output)
+
+        ConjoinXRuns(InputWorkspaces=self.output,
+                     SampleLogAsXAxis=self.scan_parameter,
+                     FailBehaviour="Stop",
+                     OutputWorkspace=self.output)
+
+        ExtractUnmaskedSpectra(InputWorkspace=self.output, OutputWorkspace=self.output)
+
+        Transpose(InputWorkspace=self.output, OutputWorkspace=self.output)
```

Comment:
I think we decided that transpose should be optional with a flag.

Reply:
However the default should be to transpose, since for diffractograms it makes more sense to see the x-axis as Q.